### PR TITLE
Azure Virtual Machine Support

### PR DESCRIPTION
**Pull request by bryan-bar**
Azure Virtual Machine Support Changes:
* terraform version >= 1.3.6 required
* agreement resource added to confirm operating system agreements automatically
* var.additional_volumes created in machine module so that validation can be used
* added support for zones
    * List of regions with zones: https://learn.microsoft.com/en-us/azure/reliability/availability-zones-service-support
    * use of zoned virtual machines requires a zoned public ip with standard sku/static ip or results in error due to not being in the same
* use of regions without zones supported
  * zone is passed in as null to the machine module when assigned to 0
* firewall support for public access and regional subnet access
* region peering support
* validations a mix of variable validation, lifestyle preconditions and provider's default checks
  * Preferably any invalid configurations are caught during `terraform plan` but provider implementation varies
  * `contains` fails during `terraform plan` if given a null value
      * use `try(contains(...), boolean)` as a workaround when null value needs to be checked when not null
        * shortcircuit evalution does not work here: `variable == null || contains([...], variable)` will fail
* provider resource uses `prevent_deletion_if_contains_resources = false`
  * https://github.com/hashicorp/terraform-provider-azurerm/issues/16155
  * https://github.com/hashicorp/terraform-provider-azurerm/issues/16928
  * Needed since destruction can fail due to:
      * Virtual machine can fail to start after provisioning and provider loses track of the resource since it eventually fails, leaving the vm behind in the failed state
      * NICs deleted too quickly and asks to retry after 180 seconds

### CODE REVIEW

**Review by jt-edb — COMMENTED** [review]
Nice work! Only 2 minor comments.

File: edbterraform/data/templates/azure/providers.tf.j2 [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    resource_group {
+      https://github.com/hashicorp/terraform-provider-azurerm/issues/16155
+      https://github.com/hashicorp/terraform-provider-azurerm/issues/16928
+      Without the following setting, destruction can fail due to:
+      * Virtual machine status failed and provider loses track of resources
+      * NICs deleted too quickly and asks for retry after 180 seconds
+      
+      prevent_deletion_if_contains_resources = false
+    }
+    */
```
[the remark this answers is not in the document]
Comment (bryan-bar) [reply edbterraform/data/templates/azure/providers.tf.j2]: @jt-edb ✅

File: edbterraform/data/templates/azure/network.tf.j2 [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  providers = {
+    azurerm = azurerm.{{ region_ }}
+  }
+}
+
+/*
+module "routes_{{ region_ }}" {
+  source = "./modules/routes"
+}
+*/
```
[the remark this answers is not in the document]
Comment (bryan-bar) [reply edbterraform/data/templates/azure/network.tf.j2]: @jt-edb ✅

**Review by jt-edb — APPROVED** [review]
LGTM

**Pull request merged**